src/utils.jl

```diff
@@ -180,5 +180,85 @@ makesym(t::Symbolic; kwargs...) = Sym{symtype(t)}(tosymbol(t; kwargs...))
 makesym(t::Num; kwargs...) = makesym(value(t); kwargs...)
 
 var_from_nested_derivative(x, i=0) = (missing, missing)
-var_from_nested_derivative(x::Term,i=0) = operation(x) isa Differential ? var_from_nested_derivative(arguments(x)[1],i+1) : (x,i)
-var_from_nested_derivative(x::Sym,i=0) = (x,i)
+var_from_nested_derivative(x::Term,i=0) = operation(x) isa Differential ? var_from_nested_derivative(arguments(x)[1], i + 1) : (x, i)
+var_from_nested_derivative(x::Sym,i=0) = (x, i)
+
+function degree(p::Sym, sym=nothing)
+    if sym === nothing
+        return 1
+    else
+        return p.name == sym.name ? 1 : 0
+    end
+end
+
+function degree(p::Pow, sym=nothing)
+    if sym === nothing
+        return p.exp
+    else
+        if hasproperty(p.base, :name)
+            p_name = p.base.name
+        elseif hasproperty(p.base, :f)
+            p_name = p.base.f
+        else
+            p_name = nothing
+        end
+        if hasproperty(sym, :name)
+            sym_name = sym.name
+        elseif hasproperty(sym, :f)
+            sym_name = sym.f
+        else
+            sym_name = nothing
+        end
+        return p_name == sym_name && (p_name != nothing) && (sym_name != nothing) ? p.exp : 0
+    end
+end
+
+function degree(p::Add, sym=nothing)
+    dict = p.dict
+    ks = collect(keys(dict))
```

Comment:
You can iterate on `keys(dict)`. Calling `collect` is unnecessary and allocating.